### PR TITLE
LaunchDarkly documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Documentation for the LaunchDarkly platform.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: destinations/testing-and-personalization/launchdarkly.mdx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    ip: "10.11.23.22",
+    lastName: "User",
+    name: "Test User",
+    privateAttribureNames: ["avatar", "country"],
+    secondary: "abcd21234"
+  }
+);
+```
+
+The following table lists the supported fields for the `identify` call:
```
Comment (saikumarrs) [inline destinations/testing-and-personalization/launchdarkly.mdx:77]: This is more appropriate.
```suggestion
The following table lists the supported fields for the `traits` object in the `identify` call:
```

File: destinations/testing-and-personalization/launchdarkly.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+| Field | Type | Presence | Description | 
+|:------|:-----|:---------|:-------------|
+| `key`    | String | Required | Unique user identifier |
```
Comment (saikumarrs) [inline destinations/testing-and-personalization/launchdarkly.mdx:81]: @ItsSudip `key` field is as such not sent in the identify call's traits but deduced as explained below.

CC @ameypv-rudder

File: destinations/testing-and-personalization/launchdarkly.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+If you have chosen to bucket the users by a specific attribute, the <code class="inline-code">secondary</code> field can be used to further distinguish users who are otherwise identical as per that attribute. 
+</div>
+
+### key
+
+As mentioned in the above table, the `key` field uniquely identifies a user.
+
+Some important cases to consider pertaining to the `key` field are listed below:
+
+- `key` is the only required field in the `identify` call, where RudderStack maps the unique user identifier (user ID). If no unique identifier is provided, RudderStack will automatically assign `anonymousId` as `key`.
```
Comment (saikumarrs) [inline destinations/testing-and-personalization/launchdarkly.mdx:105]: Needs more clarification.
```suggestion
- `key` is the only required field in the Launch Darkly's `identify` call, where RudderStack maps the unique user identifier (user ID). If no unique identifier is provided, RudderStack will automatically assign `anonymousId` as `key`.
```

File: destinations/testing-and-personalization/launchdarkly.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+- If you omit `key` and also set `anonymous` as `true`, the SDK will generate a UUID string and use that as the value for `key`. It also attempts to persist that value in local storage if possible, so that the next anonymous user gets the same `key`. However, in cases where the local storage is unavailable, a new `key` is generated every time the user is specified.
```
Comment (saikumarrs) [inline destinations/testing-and-personalization/launchdarkly.mdx:107]: @ItsSudip This whole paragraph is not accurate.
CC @ameypv-rudder

File: destinations/testing-and-personalization/launchdarkly.mdx (hunk 2 above)
Comment (ItsSudip) [reply destinations/testing-and-personalization/launchdarkly.mdx:81]: yes

File: destinations/testing-and-personalization/launchdarkly.mdx (hunk 4 above)
Comment (ItsSudip) [reply destinations/testing-and-personalization/launchdarkly.mdx:107]: Yes we don't have to provide key in identify call. We are mapping userId to key. If userId is not there then we are mapping anonymousId to key.

Comment (saikumarrs) [reply destinations/testing-and-personalization/launchdarkly.mdx:107]: We should remove this.

File: destinations/testing-and-personalization/launchdarkly.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply destinations/testing-and-personalization/launchdarkly.mdx:81]: Removed it from the table.

File: destinations/testing-and-personalization/launchdarkly.mdx (hunk 4 above)
Comment (ameypv-rudder) [reply destinations/testing-and-personalization/launchdarkly.mdx:107]: Sounds good - this is removed.

File: destinations/testing-and-personalization/launchdarkly.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply destinations/testing-and-personalization/launchdarkly.mdx]: agreed - this is incorporated.

**Review by saikumarrs — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]
LGTM

**Pull request merged**